mongo/integration/unified_spec_test.go

```diff
@@ -337,18 +337,12 @@ func runOperation(mt *mtest.T, testCase *testCase, op *operation, sess0, sess1 m
 		mt.Fatalf("unrecognized operation object: %v", op.Object)
 	}
 
-	// ensure error occurred and it's the error we expect
```

Comment:
Our code didn't call `verifyError` if `op.Result` was nil because some tests specify that an error occurred, but don't provide any details about which error that is. In this case, we would only check that the error was non-nil and return successfully.

Instead of modifying both the code here and in `verifyError` to ignore `mongo.ErrUnacknowledgedWrite`, I made these changes so we always call `verifyError` if an error was expected. If `op.Result` is nil, we pass in an empty non-nil `*operationError`.